### PR TITLE
fix: [Cherry-pick] Disk resource is not requested for index loaded with disk (#30757)

### PR DESCRIPTION
**Pull request by congqixia**
Cherry pick from master
pr: #30757
See also #30756

This PR:
- Request disk resource when index type, version loaded with disk
- Add attribute cache for index utility
- Add `typeutil.Pair`

---------

### CODE REVIEW

File: internal/querynodev2/segments/index_attr_cache.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			var result bool
+			GetDynamicPool().Submit(func() (any, error) {
+				cIndexType := C.CString(indexType)
+				defer C.free(unsafe.Pointer(cIndexType))
+				cEngineVersion := C.int32_t(indexInfo.GetCurrentIndexVersion())
+				isLoadWithDisk = bool(C.IsLoadWithDisk(cIndexType, cEngineVersion))
+				return nil, nil
+			}).Await()
+			c.loadWithDisk.Insert(typeutil.NewPair[string, int32](indexType, engineVersion), result)
+			return result, nil
```
Comment (weiliu1031) [inline internal/querynodev2/segments/index_attr_cache.go:74]: value in `loadWithDisk` always be false

Comment (congqixia) [reply internal/querynodev2/segments/index_attr_cache.go:74]: fixed

**Pull request merged**